packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -528,7 +546,7 @@ const DataGrid = forwardRef((props, ref) => {
       onCellRangeSelect(cellCoordinates.row, cellCoordinates.col, event.keyCode);
     }
 
-    if (nextRow >= gridRowCount || nextCol >= displayedColumns.length) {
+    if (nextRow >= gridRowCount || (!isSection(nextRow) && nextCol >= grid.current.rows[nextRow].cells.length)) {
```

Comment:
The change to add the isSection check does not seem correct.

Reply:
Adding this is necessary to allow for navigation between sections as the nextRow cells length in the grid ref for it would be 1 and navigation won't be allowed to the right sections and columns across sections.